### PR TITLE
[DI] Fix tracking of source class changes for lazy-proxies

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -398,6 +398,7 @@ private function generateProxyClasses()
             if (!$proxyDumper->isProxyCandidate($definition)) {
                 continue;
             }
+            $this->container->getReflectionResource($definition->getClass());
```
Comment (sroze) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: Where does this method come from and what is it going? 🤔

Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: From the moon of course! comment added :)

Comment (sroze) [reply src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: That's weird, I can't find its definition anywhere. Maybe only on GitHub's moon version? 😄 

Though, registering something by calling a getter is weirdo.

Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: oups, bad method name :) fixed

Comment (sroze) [reply src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: Aaaah, makes much more sense!

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 2]
```diff
@@ -842,7 +842,7 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
         $dumper = new PhpDumper($container);
 
         if (class_exists('ProxyManager\Configuration') && class_exists('Symfony\Bridge\ProxyManager\LazyProxy\PhpDumper\ProxyDumper')) {
-            $dumper->setProxyDumper(new ProxyDumper(substr(hash('sha256', $cache->getPath()), 0, 7)));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Kernel.php]: this doesn't serve any purpose now, and made container's cache path-dependend

**Review by stof — APPROVED** [review]

File: src/Symfony/Component/HttpKernel/Kernel.php (hunk 2 above)
Comment (stof) [reply src/Symfony/Component/HttpKernel/Kernel.php]: should we deprecate the salt argument in 4.1 ?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/Kernel.php]: that's possible, although it doesn't really hurt, so low priority on my side (if anyone wants to do it, PR welcome :))

**Review by Tobion — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**